lib/src/io_utils.dart

```diff
@@ -66,7 +66,7 @@ final newLinePartOfRegexp = new RegExp('\npart of ');
 
 final RegExp quotables = new RegExp(r'[ "\r\n\$]');
 
-/// Best used with Future<void>.
+/// Best used with Future<Null>.
 class MultiFutureTracker<T> {
```

Comment:
Not for this PR, but there is a pool package (https://pub.dartlang.org/documentation/pool/1.3.4/), which may also do what you want here.